app/views/devise/forms/_users_form.erb

```diff
@@ -0,0 +1,50 @@
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: {method: resource.new_record? ? :post : :put}) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+  <div class="form-group">
+    <%= f.label :email %>
+    <%= f.email_field :email, autofocus: true, autocomplete: 'email', class: 'form-control' %>
+  </div>
+
+  <% unless resource.new_record? %>
+    <% if devise_mapping.confirmable? && resource.pending_reconfirmation? %>
+      <div><%= t('devise.registrations.edit.currently_waiting_confirmation_for_email', email: resource.unconfirmed_email) %></div>
+    <% end %>
+  <% end %>
+
+  <div class="form-group">
+    <%= f.label :password %>
+    <% if @minimum_password_length %>
+      <small class="form-text text-muted"><%= t('devise.shared.minimum_password_length', count: @minimum_password_length) %></small>
+    <% end %>
+    <%= f.password_field :password, autocomplete: 'new-password', class: 'form-control' %>
+  </div>
+
+  <div class="form-group">
+    <%= f.label :password_confirmation %>
+    <%= f.password_field :password_confirmation, autocomplete: 'new-password', class: 'form-control' %>
+  </div>
+
+  <% unless resource.new_record? %>
+    <div class="form-group">
+      <%= f.label :current_password %>
+      <%= f.password_field :current_password, autocomplete: 'current-password', class: 'form-control' %>
+      <small class="form-text text-muted"><%= t('devise.registrations.edit.we_need_your_current_password_to_confirm_your_changes') %></small>
+    </div>
+  <% end %>
+
+  <%= fields_for :location do |location_form| %>
```

Comment:
I think this form should generate something like (I'll create `user`, `user_location` and `location` all together :) )
```
User.create(email: "123@example.com", password:"passowrd", user_locations_attributes: [location_attributes: {name: "Warszawa, Polska"}])
```